lxd/instance_snapshot.go

```diff
@@ -304,8 +305,26 @@ func snapshotPut(d *Daemon, r *http.Request, sc instance.Instance, name string)
 	return operations.OperationResponse(op)
 }
 
-func snapshotGet(sc instance.Instance, name string) response.Response {
-	render, _, err := sc.Render()
+func snapshotGet(s *state.State, snapInst instance.Instance, name string) response.Response {
+	// As this is a relatively expensive operation we only provide it as a render option when querying
+	// snapshot info and not for every call to Render(). It is important that the snapshot not be mounted
+	// here as mounting a snapshot can trigger a very expensive filesystem UUID regeneration, so we rely on
+	// the driver implementation to get the info we are requesting as cheaply as possible.
```

Comment:
This means that a call to `/1.0/instances?recursion=2` will report a size of 0 then?

That's probably fine, at least for now, though then I'd argue that `Size` on a snapshot ought to be marked as `omitempty`.

I also think we probably still want the sizes reported in `/1.0/instances/NAME/snapshots?recursion=1` as that's what we'd use for `lxc info`.

Reply:
@stgraber I'm happy to add it anywhere its needed, I'll define a function that can be used to generate the option into the Render() function. Given it is likely to be a relatively expensive call (especially when used in a list context where its being used for a list of instance each possibly having several snapshots) so I thought I would start conservatively.

I'll add it to those 2 places you suggested.

RE the `omitempty` option, I considered that, but given my understanding of this field is that it should represent the CoW usage of a snapshot, I'd imagine that 0 is valid value for that size (i.e no differences from the original source volume), which is also why I initialised it to -1 so its clear when the value hasn't been set.

Is my understanding correct or have I missed something?

Reply:
@stgraber have updated those API routes to populate the snapshot size now.